aws/session/internal/ini/number_helper.go

```diff
@@ -0,0 +1,135 @@
+package ini
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws/awserr"
+)
+
+// numberHelper is used to dictate what format a number is in
+// and what to do for negative values. Since -1e-4 is a valid
+// number, we cannot just simply check for duplicate negatives.
+type numberHelper struct {
+	binary  bool
+	octal   bool
+	decimal bool
+	hex     bool
+
+	exponent bool
+	negative bool
+}
+
+func (b numberHelper) Exists() bool {
+	return b.decimal || b.binary || b.octal || b.hex || b.exponent
+}
+
+func (b numberHelper) IsNegative() bool {
+	return b.negative
+}
+
+func (b *numberHelper) Determine(c byte) error {
+	if b.Exists() {
+		return awserr.New(ErrCodeParseError, fmt.Sprintf("multiple number formats: 0%v", string(c)), nil)
+	}
+
+	switch c {
+	case 'b':
+		b.binary = true
+	case 'o':
+		b.octal = true
+	case 'x':
+		b.hex = true
+	case 'e', 'E':
+		b.exponent = true
+		b.negative = false
```

Comment:
Nit have comment here sating that resetting negative is for the exponent's value. (which could later become negative)